using_images/db_images/mariadb.adoc

```diff
@@ -0,0 +1,754 @@
+
+====
+----
++------------+------------------+
+| database() | user()           |
++------------+------------------+
+| sampledb   | root@172.17.42.1 |
++------------+------------------+
+----
+====
+
+== Creating a Database Service from a Template
+
+{product-title} provides a xref:../../dev_guide/templates.adoc#dev-guide-templates[template] to make
+creating a new database service easy. The template provides parameter fields to
+define all the mandatory environment variables (user, password, database name,
+etc) with predefined defaults including auto-generation of password values. It
+will also define both a
+xref:../../architecture/core_concepts/deployments.adoc#deployments-and-deployment-configurations[deployment
+configuration] and a
+xref:../../architecture/core_concepts/pods_and_services.adoc#services[service].
+
+The MariaDB templates should have been registered in the default *openshift*
+project by your cluster administrator during the initial cluster setup.
+ifdef::openshift-enterprise,openshift-origin[]
+See xref:../../install_config/imagestreams_templates.adoc#install-config-imagestreams-templates[Loading the Default Image Streams and Templates]
+for more details, if required.
+endif::[]
+
+There are two templates available:
+
+* `mysql-ephemeral` is for development or testing purposes only because it uses
+ephemeral storage for the database content. This means that if the database
+pod is restarted for any reason, such as the pod being moved to another node
+or the deployment configuration being updated and triggering a redeploy, all
+data will be lost.
+* `mysql-persistent` uses a persistent volume store for the database data
+which means the data will survive a pod restart. Using persistent volumes
+requires a persistent volume pool be defined in the {product-title} deployment.
+ifdef::openshift-enterprise,openshift-origin[]
+Cluster administrator instructions for setting up the pool are located
+xref:../../install_config/persistent_storage/persistent_storage_nfs.adoc#install-config-persistent-storage-persistent-storage-nfs[here].
+endif::[]
+
+You can find instructions for instantiating templates by following these
+xref:../../dev_guide/templates.adoc#dev-guide-templates[instructions].
+
+Once you have instantiated the service, you can copy the user name, password,
+and database name environment variables into a deployment configuration for
+another component that intends to access the database. That component can then
+access the database via the service that was defined.
+
+[[using-mariadb-replication]]
+== Using MariaDB Replication
+
+ifdef::openshift-origin[]
+[IMPORTANT]
+====
+Replication support provided by the MariaDB image is experimental and should not
+be used in production.
+====
+endif::[]
+
+ifdef::openshift-enterprise[]
+[NOTE]
+====
+Enabling clustering for database images is currently in Technology Preview and
+not intended for production use.
+====
```

Comment:
I believe there's a template we're supposed to be using for tech previews now:

https://mojo.redhat.com/docs/DOC-1124368

Some of us chatted about it in #3952 , so not sure how much energy wants to go into it. Up to you.